cylc/flow/config_schema.py

```diff
@@ -0,0 +1,673 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
```

Comment:
This is  the main changed file - it may be profitable for the reviewer to compare this:
```bash
git difftool upstream/master:cylc/flow/cfgspec/suite.py cylc/flow/config_schema.py
```